src/Microsoft.AspNet.Http.Extensions/StreamCopyOperation.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Http
+{
+    // FYI: In most cases the source will be a FileStream and the destination will be to the network.
+    internal static class StreamCopyOperation
+    {
+        private const int DefaultBufferSize = 1024 * 16;
+
+        internal static async Task CopyToAsync(Stream source, Stream destination, long? length, CancellationToken cancel)
+        {
+            long? bytesRemaining = length;
+            byte[] buffer = new byte[DefaultBufferSize];
```

Comment:
https://github.com/aspnet/StaticFiles/issues/91 needs to move to this repo as well.